### ISSUE TITLE
Nested serializer names disappear in drf-spectacular swagger-ui schema view when read_only or allow_null is set?

### ISSUE THREAD

**Issue by R-P-H** [tfranzel/drf-spectacular#1231]
I am using the `drf-spectacular` library together with Swagger-UI to document my Django Rest serializers (schemas).

What I have noticed is that if I set the `read_only=True` or `allow_null=True` attribute of any nested serializer fields, the name of that nested serializer disappears from the schema view in Swagger-UI. If I remove those attributes or set them to `False`, the names re-appear. Not sure why this is happening?

```
class MySerializerSerializer(serializers.ModelSerializer):

    read_only_field = MyReadOnlySerializerSerializer(read_only=True) #--> 'MyReadOnlySerializer' name is not shown
    null_field = MyNullSerializerSerializer(read_only=False, allow_null=True) #--> 'MyNullSerializer' name is not shown
    normal_field = MyOtherSerializerSerializer(read_only=False) #--> 'MyOtherSerializer" name is shown
```
![Capture](https://github.com/tfranzel/drf-spectacular/assets/6726694/3bdb79e1-001b-475f-b09e-df095d64e656)

**Comment by tfranzel** [comment]
this seems to be only display issue with SwaggerUI. The schema should be correct though. You may need to change the swagger settings:


```python
SPECTACULAR_SETTINGS = {
    "SWAGGER_UI_SETTINGS": """{
        deepLinking: true,
        displayOperationId: true,
        persistAuthorization: true,
        defaultModelExpandDepth: 2,
        defaultModelsExpandDepth: 2
    }""",
}
```

This is what I usually use. the depth vars controls how deep swagger-ui shows models by default.

**Comment by R-P-H** [comment]
Yes, just a display issue. The schema is correct besides being nameless.

The settings didn't help to show the name.

**Comment by R-P-H** [comment]
The settings are being applied, the schemas are automatically expanded but they still don't show the name.

**Comment by tfranzel** [comment]
> Yes, just a display issue. The schema is correct besides being nameless.

so just to be sure. the names you are missing in the UI are actually present in the schema, right?

I think I have seen this before and it is a display issue with swagger-ui.
